chainercv/evaluations/eval_detection_voc.py

```diff
@@ -0,0 +1,245 @@
+from __future__ import division
+
+import numpy as np
+import six
+
+
+def eval_detection_voc(
+        pred_bboxes, pred_labels, pred_scores, gt_bboxes, gt_labels,
+        gt_difficults=None,
+        min_iou=0.5, use_07_metric=False):
+    """Calculate detection metrics based on evaluation code of PASCAL VOC.
+
+    This function evaluates recall, precison and average precision with
+    respect to a class as well as mean average precision.
+    This evaluates predicted bounding boxes obtained from a dataset which
+    has :math:`N` images.
+
+    Mean average precision is calculated by taking a mean of average
+    precision for all classes which have at least one bounding box
+    assigned by prediction or ground truth labels.
+    The code is based on the evaluation code used in PASCAL VOC Challenge.
+
+    Args:
+        pred_bboxes (list of numpy.ndarray): A list of bounding boxes.
+            The index to this list corresponds to the index of the data
+            obtained from the base dataset. Length of the list is :math:`N`.
+            The element of :obj:`pred_bboxes` is coordinates of bounding
+            boxes. This is an array whose shape is :math:`(R, 4)`,
+            where :math:`R` corresponds
+            to the number of bounding boxes, which may vary among boxes.
+            The second axis corresponds to :obj:`x_min, y_min, x_max, y_max`
+            of a box.
+        pred_labels (list of numpy.ndarray): A list of labels.
+            Similar to :obj:`pred_bboxes`, its index corresponds to an
+            index for the base dataset. Its length is :math:`N`.
+        pred_scores (list of numpy.ndarray): A list of confidence scores for
+            predicted bounding boxes. Similar to :obj:`pred_bboxes`,
+            its index corresponds to an index for the base dataset.
+            Its length is :math:`N`.
+        gt_bboxes (list of numpy.ndarray): List of ground truth bounding boxes
+            whose length is :math:`N`. An element of :obj:`gt_bboxes` is a
+            bounding box whose shape is :math:`(R, 4)`. Note that number of
+            bounding boxes in each image does not need to be same as the number
+            of corresponding predicted boxes.
+        gt_labels (list of numpy.ndarray): List of ground truth labels which
+            are organized similarly to :obj:`labels`.
+        gt_difficults (list of numpy.ndarray): List of boolean arrays which
+            is organized similarly to :obj:`labels`. This tells whether the
+            corresponding ground truth bounding box is difficult or not.
+            By default, this is :obj:`None`. In that case, this function
+            consider all bounding boxes to be not difficult.
+        min_iou (float): A prediction is correct if its Intersection over
+            Union with the ground truth is above this value.
+        use_07_metric (bool): Whether to use Pascal VOC 2007 evaluation metric
```

Comment:
You are using `PASCAL VOC` above.  Actually, both `Pascal VOC` and `PASCAL VOC` exist in the official document. However, I think it is better to unify the name in ChainerCV.